stellarphot/core.py

```diff
@@ -45,7 +45,7 @@ def __init__(self, gain=1.0 * u.electron / u.adu,
                  read_noise=1.0 * u.electron,
                  dark_current=0.01 * u.electron / u.second):
         """
-        Initializes a Camera object instance.
+        Initialize a Camera object instance by passing the required parameters.
```

Comment:
Here's the thing, it appears BELOW the rest of the description of camera and above the parameters for `__init__`, it seemed to make it more readable.  Look at the documentation view and let me know.

Reply:
Interesting. Just did some digging to try to see how the auto-summary is generated.

It looks like the "Attributes" table that is generated only generates entries for a `@property` not for other attributes. Which seems dumb to me, but I guess maybe it is because those don't have some sort of documentation to go along with them.

I also re-read the numpydoc documentation more closely and it looks like the `Parameters` section should be in the class docstring, not the `__init__`, after all. I think I said the opposite. 

For attributes, if you follow https://numpydoc.readthedocs.io/en/latest/format.html#class-docstring then for any `@property` the docstring is pulled automatically and non-`@property` attributes need to be document manually.

That will mean that for now there is duplication in the documentation, because there will be an attributes section in the class docstring and another attribute summary in generated by `automodapi`, but I think that is OK for the moment as long as everything is document somewhere accurately, even if there ends up being duplication at the moment.

I ended up adding a silly test class to my local copy of `core.py` and generated the docs just to get an idea how the pieces seem to come together (see below). This confirms that the class docstring (right below the class definition) comes first, then the parameters from the `__init__`, then the auto stuff. 

So I think the way to get the parameters where they belong is to put them in the class docstring instead of in `__init__` like I thought.

```python
class Foo:
    """
    Class that does nothing

    Attributes
    ----------
    x : int
        A number.
    y
    """
    def __init__(self, a):
        """
        Parameters
        ----------
        a : int
            A number to initialize the class with.
        """
        self.x = 5
        self._y = -42
    @property
    def y(self):
        """
        The value of ``y``.
        """
        return self._y

    def do_nothing(self):
        """
        Do nothing, but with enthusiasm.
        """
        return "enthusiasm!"
```

Reply:
This is the generated documentation for `Foo`:

<img width="739" alt="Cursor_and_Foo_—_stellarphot_vunknown_dev" src="https://github.com/feder-observatory/stellarphot/assets/1147167/4b2e7448-3ca8-4ad7-9fd3-34f4155f542d">

Reply:
That numpy documentation states very explicitly to place the parameters in the __init__ docstring.  And I think from the point of view of the programmer reviewing the code, this makes sense.  I'm just mostly annoyed the constructor isn't separated more distinctly from the code.

I am going to go through the Classes which mix attributes and @property attributes by name and use the trick of listing the attribute by name only if it is a @property attribute to fix a bit of the formatting.  I push that up shortly.

Reply:
Just checked again, as I really thought the `Parameters` were supposed to come before the `Attributes`. I think this from the numpydoc docs is ambiguous: 

> The constructor (__init__) should also be documented here, the [Parameters](https://numpydoc.readthedocs.io/en/latest/format.html#params) section of the docstring details the constructor’s parameters

I thought at first it meant the same thing you do -- that the `Parameters` should be in the `__init__` -- but now I thinkt hey means the `Parameters` should be in the class docstring not in the `__init__`.

So then I went searching for an example in numpy to see what they do rather than what they say. In the [docs for ndarray](https://numpy.org/doc/1.25/reference/generated/numpy.ndarray.html#numpy-ndarray) the `Parameters` are rendered before the `Attributes`.

Took me a minute to find where they generate the docstring since `ndarray` itself is written in C. Found the docstring here: https://github.com/numpy/numpy/blob/9cba4697aac1233834a5093a53d1a6821dea3d6a/numpy/core/_add_newdocs.py#L2213

In the docstring they put the `Parameters` ahead of the `Attributes`.

Reply:
OK, so noted, changing everything back... grumble grumble grumble.  Changes made and pushed to this branch.

Reply:
Thanks! I think it was that sentence in numpydoc that made me think that parameters were supposed to go in `__init__`. The phrasing is not clear...